src/Common/src/TypeSystem/IL/ILProvider.cs

```diff
@@ -31,10 +31,18 @@ private MethodIL TryGetIntrinsicMethodIL(MethodDesc method)
             if (owningType == null)
                 return null;
 
-            if (method.Name == "UncheckedCast" && owningType.Name == "RuntimeHelpers" && owningType.Namespace == "System.Runtime.CompilerServices")
+            string methodName = method.Name;
+
```

Comment:
The `Intrinsic` attribute is meat to ensure that intrinsics are not recognized in user code by accident.